plutus-core/plutus-core/examples/PlutusCore/Examples/Data/Vec.hs

```diff
@@ -334,7 +334,7 @@ scottHead = runQuote $ do
         . TyAbs () a (Type ())
         . TyAbs () n natK
         . LamAbs () xs (mkIterTyApp () scottVec [TyVar () a, TyApp () succT $ TyVar () n])
-        $ mkIterApp ()
+        $ mkIterAppNoAnn
```

Comment:
`mkIterTyApp` shares the same annotation, `mkIterApp` doesn't and we have a separate `mkIterAppNoAnn`. Looks kinda messy. It also messes with one's reflexes: when creating an AST, one uses an explicit annotation for virtually every function/constructor except `mkIterApp` and `mkIterInst` all of a sudden.

Reply:
Indeed, the same change should be made to `mkIterTyApp`, which I overlooked.